tests/test_aliases.py

```diff
@@ -128,3 +128,18 @@ def test_subprocess_logical_operators(xonsh_execer, xonsh_builtins, alias):
     ales = make_aliases()
     ales["echocat"] = alias
     assert isinstance(ales["echocat"], ExecAlias)
+
+
+@pytest.mark.parametrize(
+    "alias",
+    [
+        "echo 'hi' | grep h",
+        "echo 'hi' > file",
+        "cat < file",
+        "COMMAND1 e>o < input.txt | COMMAND2 > output.txt e>> errors.txt",
+    ],
+)
```

Comment:
Could you please add some tests for IO redirection characters within command arguments (e.g. `echo 'h|i' | grep h`)?

Reply:
Done